nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -1357,6 +1357,13 @@ public Response clearBulletins(
             clearBulletinsForGroupRequestEntity.setComponents(writableComponentIds);
         }
 
+        // If there are no components to clear bulletins for, return success with 0 bulletins cleared
+        if (clearBulletinsForGroupRequestEntity.getComponents().isEmpty()) {
```

Comment:
It seems like this conditional should be added to the earlier check for a `null` list of Components. In that case, the absence of Components is interpreted to mean all Components in the Process Group. Handling `null` and an empty list in the same way seems more consistent. What do you think?